idb/postgres/internal/writer/writer.go

```diff
@@ -251,13 +285,17 @@ func writeAppResource(round basics.Round, resource *ledgercore.AppResourceRecord
 		}
 	}
 
-	if resource.State.Deleted {
-		batch.Queue(deleteAccountAppStmtName, resource.Addr[:], resource.Aidx)
-	} else {
-		if resource.State.LocalState != nil {
-			batch.Queue(
-				upsertAccountAppStmtName, resource.Addr[:], resource.Aidx,
-				encoding.EncodeAppLocalState(*resource.State.LocalState))
+	if resource.State.LocalState != nil && isDAOProposalQueryInExecution(*resource.State.LocalState) {
+		voting_start := resource.State.LocalState.KeyValue[VotingStart]
+		voting_end := resource.State.LocalState.KeyValue[VotingEnd]
+		if resource.State.Deleted {
+			batch.Queue(deleteAccountAppStmtName, resource.Addr[:], resource.Aidx, voting_start.Uint, voting_end.Uint)
+		} else {
+			if resource.State.LocalState != nil {
+				batch.Queue(
+					upsertAccountAppStmtName, resource.Addr[:], resource.Aidx,
+					encoding.EncodeAppLocalState(*resource.State.LocalState), voting_start.Uint, voting_end.Uint)
+			}
```

Comment:
what this whole function is suppose to do?

Reply:
It executes the INSERT query for the proposal. We are passing parameters in this function required for INSERT query.